### ISSUE TITLE
Rewrite walker without saving/restoring encounters

### ISSUE THREAD

**Issue by SimonN** [SimonN/LixD#331]
**Lix 0.9.*:** The walker moves ahead (into walls possibly even), checks things at current location, then moves back when we don't like what we have checked and restores the old encounters.

**Idea:** That is bad, you shouldn't move back. Instead, you should check things ahead, and only move a single time. Don't reset encounters manually in the code. Such a change needs good unittests. One such test could be the replay in my next post in this issue here, and I'm not sure if I want to allow that behavior or not in the first place.

The tumbler has the same problem, but the tumbler should be rewritten anyway because its code is slow, expensive and spaghetti-like. When we have rewritten both the tumbler and the walker, we can purge `forceBodyAndFootEncounters` from class `Lixxie` for good.

**Comment by SimonN** [comment]
```
$FILENAME single/lemforum/Lovely/anyway.txt
! 68 0 ASSIGN=BLOCKER 0
! 122 0 ASSIGN=BLOCKER 1
! 134 0 ASSIGN=WALKER 1
! 136 0 ASSIGN=WALKER 0
! 239 0 ASSIGN=CLIMBER 0
! 268 0 ASSIGN=CUBER 1
```

This climber begins climbing within solid ground. It's probably okay, but we should decide on this behavior. It looks natural enough to me. Still, this behavior might be a symptom of (walker moves first, checks stuff, then moves back or climbs) instead of (walker checks first before moving at all).

**Comment by SimonN** [comment]
I've refactored the walker for 0.10.13 to use no encounters. That introduced a [physics regression](https://www.lemmingsforums.net/index.php?topic=6073.msg100803#msg100803) and I've fixed that regression for 0.10.16.

Above replay (from May 10, 2018) plays the same in 0.9.48 and in 0.10.17. Therefore, I change my mind from that post: It's not a symptom of moving first, then moving back and resetting encounters. I deem those physics reasonably natural, even though they can surprise. If those physics ever start bothering somebody, we should file them as a new bug.
